src/filesystem.cpp

```diff
@@ -45,7 +45,11 @@ namespace {
 bool do_mkdir(std::string const& path, int const mode)
 {
     (void)mode; //not used on windows
+#ifdef _MSC_VER
```

Comment:
It should be using `_mkdir` function on Windows platform, according to MSDN document [here](https://msdn.microsoft.com/en-us/library/2fkk4dzw(v=vs.80).aspx). When using `mkdir` function, Visual Studio gives a compile error, tells me that the function is deprecated, and I should use `_mkdir`. I only changed the codepath for MSVC, and left MinGW code as is.

Stackoverflow related question: [here](http://stackoverflow.com/questions/10402499/mkdir-c-function)